### ISSUE TITLE
Impeller crashes iOS app with Bad Memory Access when using camera with overlay

### ISSUE THREAD

**Issue by konsultaner** [flutter/flutter#151945]
### Steps to reproduce

- I extracted a reproducer. 
- add ios platform
- add `<key>NSCameraUsageDescription</key><string>This app needs camera access to  scan QR codes</string>` to `ios/Runner/Info.plist`
- When running tap on the qr code:

![image](https://github.com/user-attachments/assets/46c28a2d-cb37-432a-919a-14fe8a8b6160)

[reproducer.zip](https://github.com/user-attachments/files/16280698/reproducer.zip)

if I disable impeller the app shows the camera

### Expected results

Should not crash

### Actual results

![image](https://github.com/user-attachments/assets/ac84a6b2-72f5-41bb-82f1-70311f8df9d5)

### Code sample

[reproducer.zip](https://github.com/user-attachments/files/16280698/reproducer.zip)

### Screenshots or Video

![image](https://github.com/user-attachments/assets/ac84a6b2-72f5-41bb-82f1-70311f8df9d5)

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.2, on macOS 14.5 23F79 darwin-arm64, locale de-DE)
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.2)
[✓] IntelliJ IDEA Community Edition (version 2024.1.4)
[✓] Connected device (6 available)            
    ! Error: Browsing on the local area network for mbAppleWatch4. Ensure the device is unlocked and discoverable via Bluetooth. (code -27)
[✓] Network resources
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @konsultaner, a hotfix was recently released (3.22.3). I just tested your code sample and when I tapped the button, it did not crash. Can you upgrade and check if you still experience this issue?

**Comment by konsultaner** [comment]
updating to 3.22.3 works!
